King-Benx/news-watchtower-5
# offline, storage, refactor [Delivers #164335011]

https://www.pivotaltracker.com/story/show/164335011

**Overview**
In order to implement offline functionality, the application required local storage and in order to be able to improve ui/ux a loader and dialogs were required.

This PR implements local Storage, progress bar, offline functionality and a refactor of code.

**Acceptance Criteria**
- Application should operate as expected.
- A user should be able to add stories to local storage.
- A user should be able to share or delete these stored stories.
- A progress bar should appear when the application is fetching data.
- AlertDialogs should show up if there is no internet.

**How to manually test**
- Checkout to branch ft-implement-storage-164335011
- Clone project and rebuild it ```./gradlew build````
- Start the application on an emulator or device, you should be able see a progress spinner as data is being loaded
- Make sure your internet connection is turned on, tap the storage button and observe.
- Select your stored stories from the navigation drawer.
- You should be able to see stored stories.
- Close the application
- Turn off internet connection.
- Start the application again.
- An alert dialog will show up and if internet connection is not turned on, you will be redirected to your stored stories. 

**Story**

[#164335011]

**Screenshot**
<img width="120" alt="screenshot 2019-03-03 at 16 37 32" src="https://user-images.githubusercontent.com/17830204/53695945-cdff5500-3dd2-11e9-8ce6-e02a19824f58.png">
